web/fluidity.money/src/data/ethereum/prizePool.ts

```diff
@@ -0,0 +1,91 @@
+import { JsonRpcProvider, Provider } from "@ethersproject/providers";
+import { BigNumber } from "ethers";
+import { Contract, ContractInterface } from "ethers";
+import BN from "big.js";
+
+import PrizePoolABI from "./prizePoolABI.json";
+
+type ErrorType = {
+  data: { message: string };
+} & { message: string };
+
+const handleContractErrors = async (
+  error: ErrorType,
+  provider: Provider | undefined
+) => {
+  const msg = error?.data?.message ?? error?.message;
+
+  if (!msg) throw new Error(`Unknown Error: ${error}`);
+
+  // check for denial separately (these don't contain an error code for some reason)
+  if (msg === "MetaMask Tx Signature: User denied transaction signature.") {
+    throw new Error(`Transaction Denied`);
+  }
+
+  try {
+    // check if we've got a different metamask error
+    const metaMaskError = JSON.parse(msg.match(/{.*}/)?.[0] || "");
+
+    if (metaMaskError?.value.code === -32603) {
+      throw new Error(
+        `Failed to make swap. Please reset your Metamask account (settings -> advanced -> reset account)`
+      );
+    } else if (metaMaskError?.value.code === -32000) {
+      throw new Error(`Failed to make swap. Gas limit too low.`);
+    }
+
+    // otherwise, check for a 'non intrinsic' gas error (gas exhausted)
+    const { hash } = metaMaskError || {};
+    const receipt = await provider?.getTransactionReceipt(hash);
+    // found revert opcode, assume it's a gas error since we can't call this with an insufficient balance within the application
+    if (receipt?.status === 0) {
+      throw new Error(
+        `Failed to make swap. Gas limit of ${receipt?.gasUsed?.toNumber()} exhausted!`
+      );
+    }
+  } catch (e) {
+    // otherwise, use a generic error
+    throw new Error(`Failed to make swap. ${msg}`);
+  }
+};
+
+type PrizePool = {
+  amount: BigNumber;
+  decimals: number;
+};
+
+// Returns total prize pool from aggregated contract
+export const getEthTotalPrizePool = async (
+  rpcUrl: string
+): Promise<number> => {
+  const provider = new JsonRpcProvider(rpcUrl);
```

Comment:
probably want to do this in a loader (i.e. `useCache`) so that the RPC URL isn't public in the client.